bliss/api.py

```diff
@@ -234,20 +234,33 @@ def fullcat_to_astropy_table(est_cat: FullCatalog):
             on_vals[k] = v[galaxy_params_mask].reshape(-1, v.shape[-1]).cpu()
         else:
             on_vals[k] = v[is_on_mask].cpu()
+    # Split to different columns for each band
+    for b, bl in enumerate("ugriz"):
+        on_vals[f"star_log_fluxes_{bl}"] = on_vals["star_log_fluxes"][..., b]
+        on_vals[f"star_fluxes_{bl}"] = on_vals["star_fluxes"][..., b]
+    # Remove star_fluxes and star_log_fluxes
+    on_vals.pop("star_fluxes")
+    on_vals.pop("star_log_fluxes")
     n = is_on_mask.sum()  # number of (predicted) objects
     rows = [{k: v[i].cpu() for k, v in on_vals.items()} for i in range(n)]
 
     # Convert to astropy table
     est_cat_table = Table(rows)
     # Convert all _fluxes columns to u.Quantity
-    log_nmgy = (u.LogUnit(u.nmgy),)
-    est_cat_table["star_log_fluxes"] = est_cat_table["star_log_fluxes"] * log_nmgy
-    est_cat_table["star_fluxes"] = est_cat_table["star_fluxes"] * (u.nmgy,)
+    log_nmgy = u.LogUnit(u.nmgy)
+    nmgy = u.nmgy
+    for bl in "ugriz":
+        est_cat_table[f"star_log_fluxes_{bl}"].unit = log_nmgy
+        est_cat_table[f"star_fluxes_{bl}"].unit = nmgy
 
     # Create inner table for galaxy_params
```

Comment:
Why do we create a separate table for galaxy parameters? Can't these just be columns in the previous table?

Reply:
it's actually because for some reason, i cannot have an astropy column also be an (inner) astropy table.

```sh
>>> from astropy.table import Table
>>> inner_table = Table()
>>> outer_table = Table()
>>> outer_table.add_column(inner_table)
>>> outer_table
<Table length=0>
  col0 
float64
-------
>>> outer_table["col0"]
<Column name='col0' dtype='float64' length=0>

>>> outer_table.show_in_notebook()
<IPython.core.display.HTML object>
>>> outer_table["col0"].show_in_notebook()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Column' object has no attribute 'show_in_notebook'
>>> inner_table.show_in_notebook()
<IPython.core.display.HTML object>
```

Reply:
So we could either keep this two-table format, or just split `galaxy_params` into 11 individual columns and combine all into one table.

Reply:
I think separate columns in one table is better. We're moving the galaxy fluxes out of `galaxy_params`, so we would need to move those to the main table anyway, and then we might as well just move the others to keep it as one clean table.

Reply:
Done